python/utils/LinkedLibraryHolder.h

```diff
@@ -46,6 +46,12 @@ struct RuntimeTarget {
 class LinkedLibraryHolder {
 public:
 protected:
+  /// @brief Lazy-loader functor for CircuitSimulators
+  using SimulatorCreator = std::function<nvqir::CircuitSimulator *()>;
```

Comment:
nit: can we use `llvm::function_ref` here and just use singleton instances of the function objects?

Reply:
Good feedback. Updated to get rid of the stored std::function. let me know what you think.